ideascube/static/ideascube/js/ideascube.js

```diff
@@ -265,3 +265,26 @@ ID.confirmClick = function (selector) {
     };
     el.addEventListener('click', ask, false);
 };
+
+
+ID.initWifiList = function(item_selector, popup_selector, urlroot) {
+    var elements = document.querySelectorAll(item_selector);
+
+    for (var element of elements) {
+        var known = element.getAttribute('data-known') === 'True';
+        var secure = element.getAttribute('data-secure') === 'True';
+
+        if (known || !secure) {
+            var ssid = element.getAttribute('data-ssid');
+            element.setAttribute('href', urlroot + ssid);
+        } else {
+            element.setAttribute('href', popup_selector);
+
+            element.addEventListener('click', function(evt) {
```

Comment:
This would deserve a generic calling method, that would take the popup element selector as argument and a callback at minimum. So we can reuse it elsewhere.